### PR TITLE
Add newly boards (Deneyap Kart 1A v2, Deneyap Mini v2)

### PR DESCRIPTION
**Pull request by zeynepdicle**
## Description of Change
Add newly boards
- Deneyap Mini v2 (ESP32-S2)
- Deneyap Kart 1A v2 (ESP32-S3)

Update pin_arduino.h and boardds.txt files for Deneyap Boards
- Deneyap Kart (ESP32)
- Deneyap Kart 1A (ESP32)
- Deneyap Mini (ESP32-S2)
- Deneyap Kart G (ESP32-C3)

## Tests scenarios
 I have tested my Pull Request on Arduino-esp32 core v2.0.5 v2.0.4 with ESP32 and [Deneyap-Arduino-core](https://github.com/deneyapkart/deneyapkart-arduino-core) v1.3.9 with this scenarios.

## Related links
Official website: https://deneyapkart.org/en/
Documents: https://docs.deneyapkart.org/en/
Forum: https://forum.deneyapkart.org/
Github: https://github.com/deneyapkart

(*eg. Closes #number of issue*)

### CODE REVIEW

**Comment by P-R-O-C-H-Y** [comment]
Hi @zeynepdicle , I saw some recent changes in previous closed PR. Are these changes final for now? :)

**Comment by zeynepdicle** [comment]
> Hi @zeynepdicle , I saw some recent changes in previous closed PR. Are these changes final for now? :)

Hi @P-R-O-C-H-Y , yes these are final changes :)

File: variants/deneyapkart/pins_arduino.h [hunk 1]
```diff
@@ -99,13 +86,10 @@ static const uint8_t CAMXC = 32;
...
-static const uint8_t PWM1 = 22;
-
 static const uint8_t MICD = 12;
 static const uint8_t MICC = 13;
 
 static const uint8_t IMUSD = 4;
 static const uint8_t IMUSC = 15;
 
-#endif /* Pins_Arduino_h */
+#endif /* Pins_Arduino_h */
```
Comment (P-R-O-C-H-Y) [inline variants/deneyapkart/pins_arduino.h:95]: Please fix missing new line at the end of the file

File: variants/deneyapkart1A/pins_arduino.h [hunk 2]
```diff
@@ -96,12 +86,9 @@ static const uint8_t CAMXC = 32;
...
-static const uint8_t PWM0 = 23;
-static const uint8_t PWM1 = 22;
-
 static const uint8_t SDMI = 2;
 static const uint8_t SDMO = 14;
 static const uint8_t SDCS = 12;
 static const uint8_t SDCK = 27;
 
-#endif /* Pins_Arduino_h */
+#endif /* Pins_Arduino_h */
```
Comment (P-R-O-C-H-Y) [inline variants/deneyapkart1A/pins_arduino.h:94]: Please fix missing new line at the end of the file

File: variants/deneyapkartg/pins_arduino.h [hunk 3]
```diff
@@ -2,26 +2,35 @@
...
-#define RGB_BRIGHTNESS 64
-
 #define KEY_BUILTIN GPKEY
-#define BUILTIN_KEY KEY_BUILTIN
+#define BUILTIN_KEY GPKEY
 
-static const uint8_t TX = 20;
-static const uint8_t RX = 21;
+static const uint8_t TX = 21;
+static const uint8_t RX = 20;
```
Comment (P-R-O-C-H-Y) [inline variants/deneyapkartg/pins_arduino.h:33]: Are you sure the pins are right? In your docs in [this table ](https://cdn.deneyapkart.org/media/uploads/2022/12/20/dyg_detaylpinozellikleritablos.png) I see that:
```
TX = D1 = 21
RX = D0 = 20
```
but in pins_arduino.h the `D0 = 21` and `D1 = 20` which is opposite. Can you check it please?

File: variants/deneyapkartg/pins_arduino.h [hunk 4]
```diff
@@ -48,5 +57,6 @@ static const uint8_t D6 = 2;
 
 static const uint8_t PWM0 = 0;
 static const uint8_t PWM1 = 1;
+static const uint8_t PWM2 = 3;
 
-#endif /* Pins_Arduino_h */
+#endif /* Pins_Arduino_h */
```
Comment (P-R-O-C-H-Y) [inline variants/deneyapkartg/pins_arduino.h:62]: Please fix missing new line at the end of the file

File: variants/deneyapkart1Av2/pins_arduino.h [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+static const uint8_t CAMXC = 16;
+static const uint8_t CAMH  = 7;
+static const uint8_t CAMV  = 6;
+
+static const uint8_t SDMI = 14;
+static const uint8_t SDMO = 12;
+static const uint8_t SDCS = 11;
+static const uint8_t SDCK = 13;
+
+#endif /* Pins_Arduino_h */
```
Comment (P-R-O-C-H-Y) [inline variants/deneyapkart1Av2/pins_arduino.h:92]: Please fix missing new line at the end of the file

File: variants/deneyapmini/pins_arduino.h [hunk 6]
```diff
@@ -69,4 +73,10 @@ static const uint8_t D14 = 35;
...
 static const uint8_t D16 = 34;
 
-#endif /* Pins_Arduino_h */
+static const uint8_t PWM0 = 42;
+static const uint8_t PWM1 = 41;
+
+static const uint8_t DAC1 = 17;
+static const uint8_t DAC2 = 18;
+
+#endif /* Pins_Arduino_h */
```
Comment (P-R-O-C-H-Y) [inline variants/deneyapmini/pins_arduino.h:82]: Please fix missing new line at the end of the file

File: variants/deneyapminiv2/pins_arduino.h [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+static const uint8_t D13 = 0;
+static const uint8_t D14 = 33;
+
+static const uint8_t PWM0 = 42;
+static const uint8_t PWM1 = 41;
+
+static const uint8_t DAC1 = 17;
+static const uint8_t DAC2 = 18;
+
+#endif /* Pins_Arduino_h */
```
Comment (P-R-O-C-H-Y) [inline variants/deneyapminiv2/pins_arduino.h:82]: Please fix missing new line at the end of the file

File: boards.txt [hunk 8]
```diff
@@ -16504,7 +16905,7 @@ deneyapkartg.menu.PartitionScheme.default=Default 4MB with spiffs (1.2MB APP/1.5
 deneyapkartg.menu.PartitionScheme.default.build.partitions=default
 deneyapkartg.menu.PartitionScheme.defaultffat=Default 4MB with ffat (1.2MB APP/1.5MB FATFS)
 deneyapkartg.menu.PartitionScheme.defaultffat.build.partitions=default_ffat
-deneyapkartg.menu.PartitionScheme.default_8MB=8M with spiffs (3MB APP/1.5MB SPIFFS)
+deneyapkartg.menu.PartitionScheme.default_8MB=8M Flash (3MB APP/1.5MB FAT)
```
Comment (P-R-O-C-H-Y) [inline boards.txt:16908]: As this is a standard partition scheme used on many boards, lets keep the naming the same please.

File: boards.txt [hunk 9]
```diff
@@ -16527,10 +16928,10 @@ deneyapkartg.menu.PartitionScheme.huge_app.upload.maximum_size=3145728
 deneyapkartg.menu.PartitionScheme.min_spiffs=Minimal SPIFFS (1.9MB APP with OTA/190KB SPIFFS)
 deneyapkartg.menu.PartitionScheme.min_spiffs.build.partitions=min_spiffs
 deneyapkartg.menu.PartitionScheme.min_spiffs.upload.maximum_size=1966080
-deneyapkartg.menu.PartitionScheme.fatflash=16M Flash (2MB APP/12.5MB FATFS)
+deneyapkartg.menu.PartitionScheme.fatflash=16M Flash (2MB APP/12.5MB FAT)
```
Comment (P-R-O-C-H-Y) [inline boards.txt:16931]: As this is a standard partition scheme used on many boards, lets keep the naming the same please.

File: boards.txt [hunk 10]
```diff
@@ -16527,10 +16928,10 @@ deneyapkartg.menu.PartitionScheme.huge_app.upload.maximum_size=3145728
...
 deneyapkartg.menu.PartitionScheme.fatflash.build.partitions=ffat
 deneyapkartg.menu.PartitionScheme.fatflash.upload.maximum_size=2097152
-deneyapkartg.menu.PartitionScheme.app3M_fat9M_16MB=16M Flash (3MB APP/9.9MB FATFS)
+deneyapkartg.menu.PartitionScheme.app3M_fat9M_16MB=16M Flash (3MB APP/9MB FATFS)
```
Comment (P-R-O-C-H-Y) [inline boards.txt:16934]: As this is a standard partition scheme used on many boards, lets keep the naming the same please.

**Review by P-R-O-C-H-Y — CHANGES_REQUESTED** [review]
Hi @zeynepdicle, please take a look on my comments :) once everything is fixed it can be merged 👍

File: variants/deneyapkartg/pins_arduino.h (hunk 3 above)
Comment (zeynepdicle) [reply variants/deneyapkartg/pins_arduino.h:33]: > Are you sure the pins are right? In your docs in [this table ](https://cdn.deneyapkart.org/media/uploads/2022/12/20/dyg_detaylpinozellikleritablos.png) I see that:
> 
> ```
> TX = D1 = 21
> RX = D0 = 20
> ```
> 
> but in pins_arduino.h the `D0 = 21` and `D1 = 20` which is opposite. Can you check it please?

Yes, I am sure. It was wrong that i fixed.

**Comment by zeynepdicle** [comment]
Hi again @P-R-O-C-H-Y :) I fixed it, any problem?

**Review by P-R-O-C-H-Y — APPROVED** [review]
Thanks for quick fixes :) Will merge soon

**Comment by zeynepdicle** [comment]
> @zeynepdicle Thanks for quick fixes :) Will merge after CI passes.

Thank you too :)

**Pull request merged**